cpu/samd21/cpu.c

```diff
@@ -215,7 +226,7 @@ static void clk_init(void)
 #endif
 
     /* redirect all peripherals to a disabled clock generator (7) by default */
-    for (int i = 0x3; i <= 0x22; i++) {
+    for (int i = SAM0_GCLK_NUM; i <= 0x22; i++) {
```

Comment:
No, that's not correct. The loop is about peripherals (users of the GCLKs). In the code above we initialize 0 , 1 and 2. The loop initializes the remaining peripherals to use GCLK7.
The number 0x22 isn't correct either. It depends on the SAM0 variant how many peripherals there are. I don't know if there is a specific define for it. I know that SAMD21 has `I2S_1` 0x24.

Reply:
I changed it back to what is was before.
I wonder if this is necessary to begin with.